File: CS-Examples/10_Shapes/DrawOneLineThroughTwoPoints.vb
Imports Spire.Xls
Imports Spire.Xls.Core.Spreadsheet.Shapes

Namespace DrawOneLineThroughTwoPoints
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new workbook object
            Dim workbook As New Workbook()

            ' Get the first worksheet from the workbook
            Dim worksheet As Worksheet = workbook.Worksheets(0)

            ' Add a typed line shape to the worksheet and cast it to XlsLineShape
            Dim line1 As XlsLineShape = TryCast(worksheet.TypedLines.AddLine(), XlsLineShape)
            line1.LeftColumn = 3
            line1.TopRow = 3
            line1.LeftColumnOffset = 0
            line1.TopRowOffset = 0

            line1.RightColumn = 4
            line1.BottomRow = 5
            line1.RightColumnOffset = 0
            line1.BottomRowOffset = 0

            ' Add another typed line shape to the worksheet and set its start and end points
            Dim line2 As XlsLineShape = TryCast(worksheet.TypedLines.AddLine(), XlsLineShape)
            line2.StartPoint = New Point(30, 50)
            line2.EndPoint = New Point(20, 80)

            ' Specify the resulting file name for saving
            Dim result As String = "DrawOneLineThroughTwoPoints.xlsx"

            ' Save the workbook to a file in Excel 2010 format
            workbook.SaveToFile(result, ExcelVersion.Version2010)
            ' Release the resources used by the workbook
            workbook.Dispose()

            'View the document
            FileViewer(result)
		End Sub
		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace

File: CS-Examples/10_Shapes/DrawOneLineThroughTwoPoints.cs
using Spire.Xls;
using Spire.Xls.Core.Spreadsheet.Shapes;
using System;
using System.Drawing;
using System.Windows.Forms;

namespace DrawOneLineThroughTwoPoints
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            // Create a new workbook
            Workbook workbook = new Workbook();

            // Get the first worksheet
            Worksheet worksheet = workbook.Worksheets[0];

            //1)Draw a line according to relative position
            XlsLineShape line1 = worksheet.TypedLines.AddLine() as XlsLineShape;
            line1.LeftColumn = 3;
            line1.TopRow = 3;
            line1.LeftColumnOffset = 0;
            line1.TopRowOffset = 0; 

            line1.RightColumn = 4; 
            line1.BottomRow = 5; 
            line1.RightColumnOffset = 0;
            line1.BottomRowOffset = 0; 

            //2)Draw a line according to absolute position(pixels).
            XlsLineShape line2 = worksheet.TypedLines.AddLine() as XlsLineShape;
            line2.StartPoint = new Point(30, 50);
            line2.EndPoint = new Point(20, 80);

            //Save to file
            String result = "DrawOneLineThroughTwoPoints.xlsx";
            workbook.SaveToFile(result, ExcelVersion.Version2010);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //View the document
            FileViewer(result);
        }
        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}
